### PR TITLE
KAFKA-17242: MirrorCheckpointTask prints spurious timeout for startin…

### PR DESCRIPTION
**Pull request by wernerdv**
…g checkpoint and offset sync stores

Replaced the Scheduler#execute call with a single-shot asynchronous method that starts the job once but does not wait for it to complete.

### Committer Checklist (excluded from commit message)
- [x] Verify design and implementation 
- [x] Verify test coverage and CI build status
- [x] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by wernerdv** [comment]
@gharris1727 Hello, PTAL.

File: connect/mirror/src/main/java/org/apache/kafka/connect/mirror/Scheduler.java [hunk 1]
```diff
@@ -70,6 +70,10 @@ void execute(Task task, String description) {
         }
     } 
 
+    void executeAsync(Task task, String description) {
+        executor.submit(() -> executeThread(task, description));
```
Comment (C0urante) [inline connect/mirror/src/main/java/org/apache/kafka/connect/mirror/Scheduler.java]: Looking at the body of `executeThread`, won't this still cause a warning log message to be emitted if the task takes too long to execute?

```java
    private void executeThread(Task task, String description) {
        // ...
        run(task, description);
    }

    private void run(Task task, String description) {
        try {
            long start = System.currentTimeMillis();
            task.run();
            long elapsed = System.currentTimeMillis() - start;
            LOG.info("{} took {} ms", description, elapsed);
            if (elapsed > timeout.toMillis()) {
                // Warning message gets logged here
                LOG.warn("{} took too long ({} ms) running task: {}", name, elapsed, description);
            }
        }
        // ...
    }
```

**Review by C0urante — COMMENTED** [review]
Thanks @wernerdv! Had one thought, let me know what you think.

File: connect/mirror/src/main/java/org/apache/kafka/connect/mirror/Scheduler.java (hunk 1 above)
Comment (wernerdv) [reply connect/mirror/src/main/java/org/apache/kafka/connect/mirror/Scheduler.java]: Removed logging for the `executeAsync` method.
Let me know what you think.

**Review by C0urante — APPROVED** [review]
Thanks, LGTM! Will merge pending CI.

**Comment by C0urante** [comment]
@wernerdv You don't have to keep merging `trunk` into this branch. I'll let you know if I see an issue with the CI build but there's a lot of flakiness on there so don't worry if you see a failure. Better to just let the current build complete since every new commit causes any in-progress builds to be aborted, and they usually take a few hours to finish.

**Pull request merged**